src/osp/OneStepProverHostIo.sol

```diff
@@ -101,16 +101,35 @@ contract OneStepProverHostIo is IOneStepProver {
         state.u64Vals[idx] = val;
     }
 
+    uint256 constant BLS_MODULUS =
+        52435875175126190479447740508185965837690552500527637822603658699938581184513;
+    uint256 constant PRIMITIVE_ROOT_OF_UNITY =
```

Comment:
Could we provide this with the proof data, or have it be part of the machine, to avoid needing to upgrade the contracts in future if this param changes for a different version? They talk about enabling that on the EIP, but I'm not sure it's worth it vs the simplicity of hardcoding it here. What do you think?

Reply:
Hmm.. I think if the user supplies the root of unity, and we validate that `modExp256(rootOfUnity, fieldElementsPerBlob, blsModulus) == 1` we don't need to hardcode the root of unity, but personally I'd be more worried about that than just hardcoding it and upgrading it if necessary.